### ISSUE TITLE
Can't install app on other Macs

### ISSUE THREAD

**Issue by dsquared22** [mherrmann/fbs#266]
Hi - I have a paid license for fbs Pro.
 * Your operating system(s) - **Mac OS X 12.1**
 * Your Python version **3.7**
 * Your fbs version **LATEST**
 * Your PyInstaller version **LATEST**
 * Your PyQt / PySide version **Pyside6**

Everything worked fine on my Mac.  Was able to do fbs run, fbs freeze, fbs installer with no issues.  And then installed the app via the generated .dmg file on my Mac and it installs and runs fine.

However, when I copy the .dmg file to another Mac, and then try to install the app, I get this error:
![Image 1-27-22 at 10 47 PM](https://user-images.githubusercontent.com/7075948/151588960-99e4fb0d-c159-447a-a571-e995d441d7c1.jpeg).

Basically, it says "madlibs" (app name) is damaged and can't be opened.

Any suggestions?  Thanks.

**Comment by mherrmann** [comment]
Hi, what's the OS version of the other Mac? If it's lower than 12.1, then you might have to freeze and create the installer on the oldest macOS version you wish to support. From [the relevant section in fbs's Manual](https://build-system.fman.io/manual/#releasing-mac):

> Unlike Windows, there is one thing to take into account when publishing your app: Try to build it on as old a version of macOS as possible. This improves the compatibility of your app with older versions of macOS. For example, an app built on macOS 10.10 is most likely to also run on 10.14, but not the other way around. Most people use virtual machines to run old versions of macOS.

**Comment by meramsey** [comment]
To second this. 
Got a report of this same issue from someone from compiling with Monterey and latest everything pyinstaller fbs_pro.1.1.0 as well.

Reaching out to get details of what there version is and looks like I'm going to have to create a fresh MacOS dev environment from scratch again but with lower version if that is the only fix  :/

**Comment by meramsey** [comment]
@dsquared22 so this issue is not about which version its compiled with like i thought. I spent last 6 hours getting multiple MacOS KVM VM's setup to test this out.

Compiling on like High Sierra and then copying that installer over to any other version including newer OS like Monterey/BigSur showed that same misleading err um straight up lying error message. It appears that Mac is now being more sinister when your app is not codesigned.  I can install the same app on the system compiled on no problem no warning which makes it all the more frustrating till you try on other systems or have others test. I also had 2 other people test this out on various versions and they had the same issue.....

The good news is there is an easy fix but it requires the end user running an extra command in terminal after allowing the app to be installed. Thanks to the tip from [here](https://osxdaily.com/2019/02/13/fix-app-damaged-cant-be-opened-trash-error-mac/) 
```
With the command line you can use xattr to view and remove extended attributes from a file on the Mac including the application throwing the “Appname.app is damaged and can’t be opened. You should move it to the Trash.” error message.

Launch Terminal and then issue the following command:

xattr -cr /path/to/application.app

For example:

xattr -cr /Applications/Signal.app

The -c flag removes all attributes, whereas -r applies recursively for the entire targeted .app directory contents.

The xattr command can also be used to remove the ‘application downloaded from the internet’ error message on the Mac too. Again this is only recommended to advanced users because modifying extended attributes may have unintended consequences, and again you might be attempting to run an app that you should not be running, either for stability, privacy, security, or other reasons.
```

Which i tested and confirmed the app is not in fact damaged or corrupted it just doesn't meet apples signed standards and they aren't courteous enough to fess up to why there blocking it under that false pretense. I'm all for security but lying about the reason and blocking something in this manner and not having the old options they used to have to allow the app through which doesn't exist anymore like it states in their guides is all the more enraging when you gotta deal with it....

Looks like I'll have to advise the few people using Mac to do that until I can be bothered with dealing with getting an Apple Developer account and code signing license. 

Hopefully this saves someone else the hours wasted debugging and trying older and older MacOS just to find out its not at all that issue.

**Comment by mherrmann** [comment]
Sorry for your terrible experience @meramsey. I hate Apple for things like this. And getting a VM setup is also a nightmare. Your post here will likely help other people avoid many wasted hours. So thank you very much for sharing.

I have some code for code signing on macOS, but it's not yet in fbs. Once you have a Developer certificate, I could send it to you on a private channel. And then if you find some improvements to it from your use cases, I could integrate it into fbs Pro for you and others.

**Comment by dsquared22** [comment]
Thank you Michael.

David

On February 21, 2022, Michael Herrmann ***@***.***> wrote:
> Sorry for your terrible experience @meramsey
> <https://github.com/meramsey>. I hate Apple for things like this. And
> getting a VM setup is also a nightmare. Your post here will likely
> help other people avoid many wasted hours. So thank you very much for
> sharing.
>
> I have some code for code signing on macOS, but it's not yet in fbs.
> Once you have a Developer certificate, I could send it to you on a
> private channel. And then if you find some improvements to it from
> your use cases, I could integrate it into fbs Pro for you and others.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/mherrmann/fbs/issues/266#issuecomment-1046565089>,
or unsubscribe <https://github.com/notifications/unsubscribe-
auth/ABV7Q3FJDHWRL2JDMAMB7H3U4HVCDANCNFSM5NBIFOFA>.
Triage notifications on the go with GitHub Mobile for iOS
<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-
email&mt=8&pt=524675> or Android
<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-
email%26utm_medium%3Demail%26utm_source%3Dgithub>.
You are receiving this because you were mentioned.Message ID:
***@***.***>

**Comment by meramsey** [comment]
> Sorry for your terrible experience @meramsey. I hate Apple for things like this. And getting a VM setup is also a nightmare. Your post here will likely help other people avoid many wasted hours. So thank you very much for sharing.
> 
> I have some code for code signing on macOS, but it's not yet in fbs. Once you have a Developer certificate, I could send it to you on a private channel. And then if you find some improvements to it from your use cases, I could integrate it into fbs Pro for you and others.

sounds good I'll work on that and get back to you. appreciate the offer :)

**Comment by mherrmann** [comment]
@meramsey and I were in touch via email back then, but I didn't see a way to incorporate his work into fbs in a way that would be usable for others. So unfortunately, it's not in fbs yet.

**Comment by mhogg** [comment]
Any movement on this? I'm facing the same issue. I'm building on macos 12 and running on macos 14. 'xattr' works, but not ideal. Code signing on mac?
